train.py

```diff
@@ -112,16 +118,30 @@ def main(job_config: JobConfig):
     # 3. max_seq_len base on inputs
     model_config.norm_type = job_config.model.norm_type
     model_config.vocab_size = tokenizer.n_words
+    model_config.vocab_size = 50000
     model_config.max_seq_len = job_config.training.seq_len
 
     logger.info(f"Building {model_name} {job_config.model.flavor} with {model_config}")
-    with torch.device("meta"):
-        model = model_cls.from_model_args(model_config)
+    # with torch.device("meta"):
+    model = model_cls.from_model_args(model_config)
+
+    # load the model on rank 0 only, then FSDP will distribute the weights
+    if job_config.model.init_weights:
+        if dp_rank == 0:
+            # model.to_empty(device=init_device)
+            model.init_weights()
+    else:
+        if dp_rank == 0:
+            # model.to_empty(device=init_device)
+            model_name_to_weights_loading_fns[model_name](
+                model, weights_path=job_config.model.load_weights_path,
+                source=job_config.model.weights_source
+            )
 
     # a no-op hander if float8 is not enabled
-    float8_handler = Float8Handler(job_config, parallel_dims)
+    # float8_handler = Float8Handler(job_config, parallel_dims)
     # swap to Float8Linear based on float8 configs
-    float8_handler.convert_to_float8_training(model)
+    # float8_handler.convert_to_float8_training(model)
```

Comment:
I'm guessing this broke the transformers implementation somehow, but this will give us a nice improvement in training throughput. So let's either have the option to use it, or just keep in mind that we'll be bringing this back later.

Reply:
i commented this before implementing OPT, are we going to need this?

Reply:
We will need this as soon as we get the H100.